### PR TITLE
enable gosec tool

### PR DESCRIPTION
**Pull request by jabia99**
Gosec is a security tool for golang. It's already enabled in image-builder. This PR will enable this tool and solve the vulnerability issues it finds.

Gosec available rules: https://github.com/securego/gosec#available-rules 

This pull request includes:

- [ ] adequate testing for the new functionality or fixed issue
- [ ] adequate documentation informing people about the change such as
  - [ ] submit a PR for the [guides](https://github.com/osbuild/guides) repository if this PR changed any behavior described there: https://www.osbuild.org/guides/

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -3,14 +3,27 @@ linters-settings:
...
+  timeout: 5m
 
 issues:
   # Maximum issues count per one linter. Set to 0 to disable. Default is 50.
   max-issues-per-linter: 0
 
   # Maximum count of issues with the same text. Set to 0 to disable. Default is 3.
   max-same-issues: 0
+
+  exclude-rules:
```
Comment (jabia99) [inline .golangci.yml:23]: "excludes:" under linters-settings didn't work for some reason ([gosec config](https://golangci-lint.run/usage/linters/#gosec)). So i had to use text based exclusion

Comment (ondrejbudai) [reply .golangci.yml:23]: I would honestly prefer doing local excludes over global excludes. Feels safer.

File: .golangci.yml [hunk 2]
```diff
@@ -3,14 +3,27 @@ linters-settings:
...
+    - text: "G404:" # Ignore weak random number generator lint. We do not need strong randomness here.
+      linters:
+        - gosec
+    - text: "G601:" # Ignore Implicit memory aliasing in for loop
```
Comment (ondrejbudai) [inline .golangci.yml:27]: Did you check that loops that break this rule aren't buggy? They feel suspicious to me.

**Review by ondrejbudai — CHANGES_REQUESTED** [review]
Please, can you do local excludes? I think it's very valuable to have all the rules enabled for new code.

File: .golangci.yml (hunk 2 above)
Comment (jabia99) [reply .golangci.yml:27]: I checked some of them but not all. I seemed to me that unintended bugs would be really noticeable in this case.

Comment (jabia99) [reply .golangci.yml:27]: I will do local excludes in every case and review the loops

**Review by atodorov — APPROVED** [review]

**Pull request merged**